src/_nebari/stages/kubernetes_keycloak/__init__.py

```diff
@@ -72,38 +71,27 @@ class Auth0Config(schema.Base):
     auth0_subdomain: str
 
 
-class Authentication(schema.Base, ABC):
-    _types: typing.Dict[str, type] = {}
-
+class BaseAuthentication(schema.Base):
     type: AuthenticationEnum
 
-    # Based on https://github.com/samuelcolvin/pydantic/issues/2177#issuecomment-739578307
 
-    # This allows type field to determine which subclass of Authentication should be used for validation.
+class PasswordAuthentication(BaseAuthentication):
+    type: AuthenticationEnum = AuthenticationEnum.password
 
-    # Used to register automatically all the submodels in `_types`.
-    def __init_subclass__(cls):
-        cls._types[cls._typ.value] = cls
```

Comment:
Probably due to internal changes, this won't work in Pydantic 2. Instead of trying to debug this, I adopted an alternative from https://github.com/pydantic/pydantic/issues/2203